pandas/tests/test_index.py

```diff
@@ -3096,17 +3095,16 @@ def test_join_right(self):
         self.assertIsNone(ridx)
 
         # non-unique
-        """
-        idx = Index([1,1,2,5])
-        idx2 = Index([1,2,5,7,9])
+        idx = Index([1, 1, 2, 5])
+        idx2 = Index([1, 2, 5, 7, 9])
         res, lidx, ridx = idx.join(idx2, how='right', return_indexers=True)
-        eres = idx2
-        elidx = np.array([0, 2, 3, -1, -1])
-        eridx = np.array([0, 1, 2, 3, 4])
+        eres = Index([1, 1, 2, 5, 7, 9])  # 1 is in idx2, so it should be x2
+        elidx = np.array([0, 1, 2, 3, -1, -1])
+        eridx = np.array([0, 0, 1, 2, 3, 4])
         self.assertTrue(res.equals(eres))
         tm.assert_numpy_array_equal(lidx, elidx)
         tm.assert_numpy_array_equal(ridx, eridx)
-
+        """
```

Comment:
ok, see what they do (and of course if they make sense). I suspect that the code is actually working, so adjust the tests if you would.